client/components/common/SelectElement/SelectActivity.vue

```diff
@@ -32,33 +34,37 @@
         </v-btn>
       </template>
     </v-treeview>
-    <v-alert :value="!hasSearchResults" color="primary" dark>
-      No matches found.
+    <v-alert :value="!!noResultsMessage" color="primary" dark>
+      {{ noResultsMessage }}
     </v-alert>
   </div>
 </template>
 
 <script>
 import groupBy from 'lodash/groupBy';
 import { isEditable } from 'shared/activities';
-import { mapGetters } from 'vuex';
+import map from 'lodash/map';
 import pluralize from 'pluralize';
 import { toTreeFormat } from 'utils/activity';
 
 export default {
   name: 'select-activity',
   props: {
-    selectedElements: { type: Array, default: () => [] }
+    selectedElements: { type: Array, default: () => [] },
+    activities: { type: Array, default: () => [] }
   },
   data: () => ({ search: '' }),
   computed: {
-    ...mapGetters('repository', ['activities']),
     groupedSelection: vm => groupBy(vm.selectedElements, 'outlineId'),
+    expandedActivityIds: vm => map(vm.activities, 'id'),
     activityTree: vm => toTreeFormat(vm.activities, []),
-    hasSearchResults() {
-      if (!this.search || !this.$refs) return true;
-      const { excludedItems, nodes } = this.$refs.treeview;
-      return excludedItems.size !== Object.keys(nodes).length;
+    noResultsMessage() {
+      const { activities, search, $refs } = this;
+      if (!activities.length) return 'Empty repository';
+      if (!search || !$refs) return '';
+      const { excludedItems, nodes } = $refs.treeview;
+      if (excludedItems.size !== Object.keys(nodes).length) return '';
```

Comment:
Would assigning this condition to boolean variable make it a bit simpler to read?
I.e.
```suggestion
      const hasSearchResults = excludedItems.size !== Object.keys(nodes).length;
      if (excludedItems.size !== Object.keys(nodes).length) return '';
```